[ARRISEOS-41318] turn on pause for buffering mechanism

przemyslaw-gorszkowski-red opened the pull request.

jacek-manko-red commented on Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamer.cpp:
```diff
@@ -1404,10 +1404,49 @@ void MediaPlayerPrivateGStreamer::handleMessage(GstMessage* message)
+                GValue item = G_VALUE_INIT;
+                switch (gst_iterator_next(iter, &item)) {
+                case GST_ITERATOR_OK: {
+                    sinkPad = static_cast<GstPad*>(g_value_get_object(&item));
+                    break;
```
done = true?
we can stop iterating after getting the 1st sink pad- there's only one anyway, right?

przemyslaw-gorszkowski-red replied:
this part was implemented by upstream

jacek-manko-red commented on Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamer.cpp:
```diff
@@ -1636,24 +1675,68 @@ void MediaPlayerPrivateGStreamer::processBufferingStats(GstMessage* message)
+            g_object_get(m_multiqueue.get(), "stats", &stats.outPtr(), nullptr);
+            const GValue* queues = gst_structure_get_value(stats.get(), "queues");
+            guint size = gst_value_array_get_size(queues);
+            for (guint i = 0; i < size; i++) {
+                guint bytes = 0;
+                if (gst_structure_get_uint(gst_value_get_structure(gst_value_array_get_value(queues, i)), "bytes", &bytes))
+                    multiqueueBufferedBytes += bytes;
+            }
```
you're counting both audio and video data here (as the same multiqueue handles both), but later you're comparing it to max-size-bytes of only one queue2 to calculate percentages- is this correct?

przemyslaw-gorszkowski-red replied:
yes, because queue2 process data which contains A/V, before qtdemux the A/V are kept as one bytes stream

przemyslaw-gorszkowski-red committed:
[ARRISEOS-41318] turn on pause for buffering mechanism

The buffering level was already corrected on Nexus with the contents of playpump.
This patch also uses multiqueue and current-level-bytes from GstQueue2 for correction,
as well as early playback resume triggering when the downloaded content
(m_maxTimeLoaded) is more than 2 sec ahead of the current playback position.
